### PR TITLE
upgrade to go 1.17

### PR DESCRIPTION
**Pull request by songleo**
https://github.com/open-cluster-management/backlog/issues/18239
Signed-off-by: Song Song Li <ssli@redhat.com>

### CODE REVIEW

File: tests/e2e.sh [hunk 1]
```diff
@@ -45,8 +45,8 @@ dex() {
 }
 
 deploy() {
-    $KUBECTL apply -f https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/setup/prometheus-operator-0servicemonitorCustomResourceDefinition.yaml
-    $KUBECTL apply -f https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/setup/prometheus-operator-0prometheusruleCustomResourceDefinition.yaml
+    $KUBECTL apply -f https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/main/manifests/setup/0servicemonitorCustomResourceDefinition.yaml
```
Comment (morvencao) [inline tests/e2e.sh:48]: others look good, just this one.
Can we just use stable version for Z stream.
Suggest to use:
```
$KUBECTL apply -f https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.9/manifests/setup/prometheus-operator-0servicemonitorCustomResourceDefinition.yaml
$KUBECTL apply -f https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.9/manifests/setup/prometheus-operator-0prometheusruleCustomResourceDefinition.yaml
```

Comment (songleo) [reply tests/e2e.sh]: got it, let me update it. thanks.

**Review by morvencao — COMMENTED** [review]
/lgtm

**Pull request merged**